code/procs/mobprocs/chatprocs.dm

```diff
@@ -544,6 +551,42 @@
 	else
 		return "[speechverb],[first_quote][font_accent ? "<font face='[font_accent]'>" : null]<span [class? class : ""]>[text]</span>[font_accent ? "</font>" : null][second_quote]"
 
+// Say emphasis stuff
+/// Transforms the speech emphasis mods from [/atom/proc/say_emphasis] into the appropriate HTML tags. Includes escaping.
+#define ENCODE_HTML_EMPHASIS(input, char, html, varname) \
+	var/static/regex/##varname = regex("(?<!\\\\)[char](.+?)(?<!\\\\)[char]", "g");\
+	input = varname.Replace_char(input, "<[html]>$1</[html]>")
+
+// Italicised text looks bad in maptext without adding phantom trailing spaces. I'm sorry. - DisturbHerb
+var/static/min_spaces_after_italics = 3 // Minimum number of trailing spaces after a portion of italicised html.
+
+// Forgive me for mixing atom and mob procs together, these need to be used in some objs - DisturbHerb
+/// Scans the input sentence for speech emphasis modifiers, notably |italics|, +bold+, and _underline_ -mothblocks
+/atom/proc/say_emphasis(input)
+	ENCODE_HTML_EMPHASIS(input, "\\|", "i", italics)
+	ENCODE_HTML_EMPHASIS(input, "\\+", "b", bold)
+	ENCODE_HTML_EMPHASIS(input, "_", "u", underline)
+
+	// Removes backslashes used to escape text modification.
+	var/static/regex/remove_escape_backlashes = regex(@"\\(_|\+|\|)", "g")
+	// Gets instances of whitespace after an italicised piece of text that contains less than 3 spaces total.
+	var/static/regex/add_spacing_after_italics = regex("((<\\/i>)(<\\/\\w>)*)(\\s{1,[min_spaces_after_italics]})(?=\\S)", "g")
+
+	input = remove_escape_backlashes.Replace_char(input, "$1")
+	var/spaces // forgive me. we don't have a helper for this.
+	for (var/i in 1 to min_spaces_after_italics)
+		spaces += " "
```

Comment:
Yeah this sucks, and string concatenation is actually kinda expensive so we do want to avoid that (especially since this is running on every said message regardless of if we have any emphasis to apply). 
My approach, drop `min_spaces_after_italics` and replace it with a var/define that just holds the exact text you want to apply (three spaces in a row, seems like). Then you can use `length()` for the `add_spacing_after_italics` regex, and just append the var/define itself to the text anywhere italics appear.

Reply:
I think this reads a little unclear, feel free to ping me for followup questions/comments